### PR TITLE
Use UpdateStatus to update status

### PR DESCRIPTION
**Pull request by Miciah**
#### Update cluster-version-operator dependency

* `Gopkg.lock`:
* `vendor/github.com/openshift/cluster-version-operator`: Import clientset.
* `vendor/github.com/google/uuid`: Update.

#### Use `UpdateStatus` to update status

Use cluster-version-operator's client's `UpdateStatus` method to update the ClusterOperator resource's `status` field; using `sdk.Update` fails since `status` was made a subresource.

* `cmd/cluster-ingress-operator/main.go`: (`main`): Get CVO client and store it in the handler.
* `pkg/stub/handler.go` (`Handler`): Add `CvoClient` field.
* `pkg/stub/status.go` (`syncOperatorStatus`): Use CVO client's `UpdateStatus` method instead of `sdk.Update`.

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -133,6 +133,14 @@
   pruneopts = "NUT"
   revision = "24818f796faf91cd76ec7bddd72458fbced7a6c1"
 
+[[projects]]
```
Comment (pravisankar) [inline Gopkg.lock]: Where is the change to Gopkg.toml?

Comment (pravisankar) [reply Gopkg.lock]: Is the Gopkg.lock out of sync with the current Gopkg.toml?

**Comment by Miciah** [comment]
/retest

Getting this error again:

    error: could not run steps: failed to wait for release pod to complete: could not wait for pod completion: pod release-latest was already deleted

File: pkg/stub/status.go [hunk 2]
```diff
@@ -71,8 +73,17 @@ func (h *Handler) syncOperatorStatus() {
 		return
 	}
 
-	if err := sdk.Update(co); err != nil {
-		logrus.Errorf("syncOperatorStatus: failed to update status of ClusterOperator %s/%s: %v",
+	c, err := clientset.NewForConfig(k8sclient.GetKubeConfig())
```
[the remark this answers is not in the document]
Comment (Miciah) [reply pkg/stub/status.go]: Thanks! Done.

**Comment by Miciah** [comment]
Latest push adds `CvoClient` to `Handler`, deletes the code to get the client in `syncOperatorStatus`, and modifies `main` to get the CVO client instead and store it in the handler.

File: Gopkg.lock (hunk 1 above)
Comment (Miciah) [reply Gopkg.lock]: I did not make any changes to `Gopkg.toml`.  I did run `dep ensure -add github.com/openshift/cluster-version-operator/pkg/generated/clientset/versioned` to get the client set.  `dep check` reports no problems after this commit.

Comment (ironcladlou) [reply Gopkg.lock]: Is this just a consequence of pruning? The constraint is the same, but the subset of files from that constraint preserved in `vendor` will change as a function of packages actually imported by code in our project.

**Comment by ironcladlou** [comment]
Makes me wonder why we don't just use generated clientsets across the board and ditch the SDK client

/lgtm

File: Gopkg.lock (hunk 1 above)
Comment (pravisankar) [reply Gopkg.lock]: We expect Gopkg.lock to be consistently generated (except for pkgs that are not pinned) from Gopkg.toml
Can you check:
```
$ mv Gopkg.lock Gopkg.lock.old
$ dep ensure
$ diff Gopkg.lock Gopkg.lock.old
# This should not show pkg differences 
```

Comment (Miciah) [reply Gopkg.lock]: @pravisankar, it fails even to generate a new `Gopkg.lock`:
```
% mv Gopkg.lock Gopkg.lock.old
% dep ensure
Solving failure: No versions of github.com/operator-framework/operator-sdk met constraints:
        v0.1.1: Could not introduce github.com/operator-framework/operator-sdk@v0.1.1, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        v0.1.0: Could not introduce github.com/operator-framework/operator-sdk@v0.1.0, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        v0.0.7: Could not introduce github.com/operator-framework/operator-sdk@v0.0.7, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        v0.0.6: Could not introduce github.com/operator-framework/operator-sdk@v0.0.6, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        v0.0.5: Could not introduce github.com/operator-framework/operator-sdk@v0.0.5, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        v0.0.4: Could not introduce github.com/operator-framework/operator-sdk@v0.0.4, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        v0.0.3: Could not introduce github.com/operator-framework/operator-sdk@v0.0.3, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        v0.0.2: Could not introduce github.com/operator-framework/operator-sdk@v0.0.2, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        v0.0.1: Could not introduce github.com/operator-framework/operator-sdk@v0.0.1, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        master: Could not introduce github.com/operator-framework/operator-sdk@master due to multiple problematic subpackages:
        Subpackage github.com/operator-framework/operator-sdk/pkg/k8sclient is missing. (Package is required by (root).)        Subpackage github.com/operator-framework/operator-sdk/pkg/util/k8sutil is missing. (Package is required by (root).)
        revert-712-decompose_gopath: Could not introduce github.com/operator-framework/operator-sdk@revert-712-decompose_gopath, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        v0.0.7: Could not introduce github.com/operator-framework/operator-sdk@v0.0.7, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
        v0.1.x: Could not introduce github.com/operator-framework/operator-sdk@v0.1.x, as it is not allowed by constraint master from project github.com/openshift/cluster-ingress-operator.
zsh: exit 1     dep ensure
dep ensure  8.23s user 7.95s system 78% cpu 20.567 total
%
```

**Pull request merged**